File: src/main/java/atm/bloodworkxgaming/oeintegration/Handler/EventHandler.java
package atm.bloodworkxgaming.oeintegration.Handler;

import atm.bloodworkxgaming.oeintegration.MainConfig;
import net.minecraft.item.ItemStack;
import net.minecraft.util.text.TextComponentString;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
import net.minecraftforge.fml.common.gameevent.TickEvent;
import oreexcavation.events.EventExcavate;
import oreexcavation.handlers.MiningAgent;

/**
 * Created by jonas on 18.06.2017.
 */
@Mod.EventBusSubscriber
public class EventHandler {
    @SubscribeEvent
    public static void onExcavateEvent(EventExcavate.Pre eventExcavate) {
        MiningAgent agent = eventExcavate.getAgent();
        ItemStack held = agent.player.getHeldItemMainhand();

        switch (IntegrationHandler.checkCanMine(held)){
            case WHITELIST:
            case MOD_DISABLED:
                break;
            case ENCHANTMENT:
                IntegrationHandler.changeToolOverwriteEnchantment(agent);
                break;
            case TINKERS_CONSTRUCT:
                IntegrationHandler.changeToolOverwriteTinkers(agent);
                break;
            case DISALLOWED:
            default:
                eventExcavate.setCanceled(true);
                if (!MainConfig.disableChatNotification){
                    agent.player.sendMessage(new TextComponentString("You need the \u00A7eExcavate Modifier \u00A7ror the \u00A76Enchantment \u00A7ron your tool to be able to excavate!"));
                }
                break;

        }
    }
}

File: src/main/kotlin/atm/bloodworkxgaming/oeintegration/handler/EventHandler.kt
package atm.bloodworkxgaming.oeintegration.handler

import atm.bloodworkxgaming.oeintegration.MainConfig
import atm.bloodworkxgaming.oeintegration.OreExcavationIntegration.LOGGER
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationHandler
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.DISALLOWED
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.ENCHANTMENT
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.MOD_DISABLED
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.TINKERS_CONSTRUCT
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.WHITELIST
import atm.bloodworkxgaming.oeintegration.integrations.IntegrationType.WHITELISTED_PACKMODE
import net.minecraft.util.text.ChatType
import net.minecraft.util.text.TranslationTextComponent
import net.minecraftforge.eventbus.api.SubscribeEvent
import oreexcavation.events.EventExcavate
import java.util.*

object EventHandler {
    @SubscribeEvent
    fun onExcavateEvent(eventExcavate: EventExcavate.Pre) {
        val agent = eventExcavate.agent

        fun cancelEvent() {
            eventExcavate.isCanceled = true
            if (!MainConfig.disableChatNotification.get()) {
                agent.player.sendMessage(
                    TranslationTextComponent("chat.oeintegration.excavate.canceled"),
                    ChatType.GAME_INFO,
                    UUID.randomUUID()
                )
            }
        }

        when (IntegrationHandler.checkCanMine(agent.player)) {
            TINKERS_CONSTRUCT -> IntegrationHandler.changeToolOverwriteTinkers(agent)
            ENCHANTMENT -> IntegrationHandler.changeToolOverwriteEnchantment(agent)
            DISALLOWED -> cancelEvent()
            MOD_DISABLED, WHITELISTED_PACKMODE, WHITELIST -> {/* All fine, continue mining */
            }
        }
    }
}
